### PR TITLE
Fixes in the code style and upylon abbrevation

### PR DESCRIPTION
**Pull request by kjawadDeveloper2**
1) Did UI fixes.
2) Solved some enum issues. 
3) Modified code architecture. 
4) Amended a flow of the first time NFT.
5) Solved the upylon issue. 
6) Renamed the routes. 
7) Solved few analysis issue. 
8) Made the price, mint , describe and choose screen only build when needed.

### CODE REVIEW

**Review by mustafapylons — APPROVED** [review]
Looks good to me.

File: lib/easel_provider.dart [hunk 1]
```diff
@@ -190,12 +189,16 @@ class EaselProvider extends ChangeNotifier {
...
     royaltyController.text = '';
+    hashtagsList.clear();
     notifyListeners();
   }
 
-  void setTextFieldValuesDescription({String? artName, String? description}) {
+  void setTextFieldValuesDescription({String? artName, String? description, String? hashtags}) {
     artNameController.text = artName ?? "";
     descriptionController.text = description ?? "";
+    if (hashtags != "") {
```
Comment (kjawadDeveloper2) [inline lib/easel_provider.dart]: You can use .isEmpty.

File: lib/models/denom.dart [hunk 2]
```diff
@@ -45,12 +47,36 @@ class Denom {
...
+                width: 14.0.h,
+                fit: BoxFit.fill,
+              );
       case kEthereumSymbol:
         return Container(padding: EdgeInsets.symmetric(vertical: 10.h), child: Image.asset(icon));
 
       default:
-        return Image.asset(icon);
+        return isTablet
+            ? Image.asset(
```
Comment (kjawadDeveloper2) [inline lib/models/denom.dart]: Revert this.

File: lib/models/denom.dart [hunk 3]
```diff
@@ -45,12 +47,36 @@ class Denom {
       case kEuroSymbol:
       case kAgoricSymbol:
       case kJunoSymbol:
-        return Image.asset(icon);
+        return isTablet
```
Comment (kjawadDeveloper2) [inline lib/models/denom.dart]: Revert this.

File: lib/repository/repository.dart [hunk 4]
```diff
@@ -215,7 +215,7 @@ class RepositoryImp implements Repository {
   @override
   Future<Either<Failure, bool>> updateNftFromDescription({required SaveNft saveNft}) async {
     try {
-      bool result = await localDataSource.updateNftFromDescription(saveNft.id!, saveNft.nftName!, saveNft.nftDescription!, saveNft.creatorName!, saveNft.step!);
+      bool result = await localDataSource.updateNftFromDescription(saveNft.id!, saveNft.nftName!, saveNft.nftDescription!, saveNft.creatorName!, saveNft.step!, saveNft.hashtags!);
```
Comment (kjawadDeveloper2) [inline lib/repository/repository.dart]: Check before force unwrapping.

File: lib/services/datasources/local_datasource.dart [hunk 5]
```diff
@@ -65,7 +65,7 @@ abstract class LocalDataSource {
   /// [String] the  name of the nft , [String] the  description of the nft
   /// [String] the  creator name of the nft , [String] the page name of the Pageview
   /// Output: [bool] returns whether the operation is successful or not
-  Future<bool> updateNftFromDescription(int id, String nftName, String nftDescription, String creatorName, String step);
+  Future<bool> updateNftFromDescription(int id, String nftName, String nftDescription, String creatorName, String step, String hashtags);
```
Comment (kjawadDeveloper2) [inline lib/services/datasources/local_datasource.dart]: Use named arguments.

File: lib/screens/splash_screen.dart [hunk 6]
```diff
@@ -79,6 +84,9 @@ class SplashScreen extends StatelessWidget {
             alignment: Alignment.center,
             child: PylonsButton(
               onPressed: () {
+                Navigator.of(context).pushNamed(RouteUtil.kRouteTutorial);
+
+                /// TODO remove the above line after testing
```
Comment (kjawadDeveloper2) [inline lib/screens/splash_screen.dart]: Leave it there till my testing completes.

File: lib/screens/splash_screen.dart [hunk 7]
```diff
@@ -34,6 +34,11 @@ class SplashScreen extends StatelessWidget {
             alignment: Alignment.center,
             child: PylonsButton(
               onPressed: () {
+
+                Navigator.of(context).pushNamed(RouteUtil.kRouteTutorial);
```
Comment (kjawadDeveloper2) [inline lib/screens/splash_screen.dart]: Leave it there till my testing completes.

Comment (ahmadhassan-dev1) [reply lib/screens/splash_screen.dart]: sure

File: lib/repository/repository.dart [hunk 8]
```diff
@@ -104,8 +104,8 @@ abstract class Repository {
   Future<Either<Failure, List<NFT>>> getNfts();
 
   /// This method will get the drafts List from the local database
-  /// Input: [id] the id of the nft which the user wants to get
-  /// Output: [NFT] returns  the List of drafts
+  /// Input: [int] the id of the nft that you want to get
+  /// Output: [NFT] returns the nft
```
Comment (kjawadDeveloper2) [inline lib/repository/repository.dart]: Wrong description.

File: lib/repository/repository.dart [hunk 9]
```diff
@@ -90,7 +90,7 @@ abstract class Repository {
   Future<Either<Failure, bool>> updateNftFromDescription({required SaveNft saveNft});
 
   /// This method will update draft in the local database from Pricing page
-  /// Input:[SaveNft] this model data contains bring [id],[tradePercentage],[price],[quantity],[step],[denomName],[isFreeDrop]
+  /// Input:[SaveNft] this model data contains bring [id],[tradePercentage],[price],[quantity],[step],[denomSymbol],[isFreeDrop]
```
Comment (kjawadDeveloper2) [inline lib/repository/repository.dart]: Update documentation.

File: lib/services/datasources/local_datasource.dart [hunk 10]
```diff
@@ -57,22 +58,23 @@ abstract class LocalDataSource {
...
 
   /// This method will get the drafts List from the local database
-  /// Output: [List][NFT] returns  the List of drafts
+  /// Input: [int] the id of the nft that you want to get
+  /// Output: [NFT] returns the nft
   Future<NFT?> getNft(int id);
 
   /// This method will update draft in the local database from description Page
   /// Input: [id] the id of the nft,
   /// [String] the  name of the nft , [String] the  description of the nft
```
Comment (kjawadDeveloper2) [inline lib/services/datasources/local_datasource.dart]: Update the documentation.

File: lib/services/datasources/local_datasource.dart [hunk 11]
```diff
@@ -57,22 +58,23 @@ abstract class LocalDataSource {
...
   /// [String] the  creator name of the nft , [String] the page name of the Pageview
   /// Output: [bool] returns whether the operation is successful or not
-  Future<bool> updateNftFromDescription(int id, String nftName, String nftDescription, String creatorName, String step);
+  Future<bool> updateNftFromDescription(SaveNft saveNft);
 
   /// This method will update draft in the local database from Pricing page
   /// Input: [id] the id of the nft, [String] the  name of the nft ,
   /// [String] the  tradePercentage of the nft , [String] the  price of the nft
```
Comment (kjawadDeveloper2) [inline lib/services/datasources/local_datasource.dart]: Update documentation.

File: lib/screens/home_screen.dart [hunk 12]
```diff
@@ -137,7 +138,7 @@ class HomeScreenContent extends StatelessWidget {
           tabletScreen: (context) => const VerticalSpace(30),
```
Comment (Amean) [inline lib/screens/home_screen.dart:138]: Should this be 30.h?

File: lib/screens/tutorial_screen.dart [hunk 13]
```diff
@@ -81,9 +83,10 @@ class _TutorialScreenState extends State<TutorialScreen> {
                           decoration: const BoxDecoration(
                             image: DecorationImage(image: AssetImage(kTooltipBalloon), fit: BoxFit.contain),
                           ),
-                          child: Text(
+                          child: const AutoSizeText(
                             kWhyAppNeeded,
-                            style: TextStyle(fontSize: 12.sp, fontWeight: FontWeight.w400, color: EaselAppTheme.kWhite),
+                            maxFontSize: 18,
+                            style: TextStyle(fontSize: 18, fontWeight: FontWeight.w400, color: EaselAppTheme.kWhite),
```
Comment (Amean) [inline lib/screens/tutorial_screen.dart:89]: Should fontSize have .sp?

File: lib/screens/home_screen.dart (hunk 12 above)
Comment (kjawadDeveloper2) [reply lib/screens/home_screen.dart:138]: @Amean  if you look inside the Veritical space its using.h

File: lib/screens/tutorial_screen.dart (hunk 13 above)
Comment (kjawadDeveloper2) [reply lib/screens/tutorial_screen.dart:89]: No AutoSizeText will handle this automatically.

**Review by Amean — APPROVED** [review]

**Pull request merged**